### ISSUE TITLE
Connection drops, frequently 'unavailable'

### ISSUE THREAD

**Issue by aegjoyce** [fredrikbaberg/hassio-addons#274]
Thanks for an excellent add-on @fredrikbaberg.

I'm having a few issues with connection stability - using the Octoprint integration in HA to connect locally to the add-on, I'm finding that the server entities often change to 'unavailable' at random.

In the HA logs I get a frequently recurring error:
```
Error requesting octoprint-b6c83dd6a764e9c3b0b935c189d4bd58 data: [Errno 104] Connection reset by peer
Error requesting octoprint-b6c83dd6a764e9c3b0b935c189d4bd58 data: Server disconnected
```
I can't see any errors in the logs in the 3DPrinter add-on at all, everything seems ok there.

Any ideas how to fix this? Unfortunately I have some automations that are meant to turn my printer off automatically after a print (if enabled) and they rely on accurate updates of the Octoprint entity states - going 'unavailable' tends to break this functionality.

**Comment by fredrikbaberg** [comment]
Thank you for using the addon!

My printer has been offline for a few months now, but I don't think I experienced this when I had it running. I won't be able to turn on the printer this month, but I'll have a look at the logs.

I assume you are using the latest version of 3DPrinter. What version of Home Assistant are you currently on? Do you know when the issue started?

**Comment by aegjoyce** [comment]
Yup latest version, and I'm on 2022.3.5 but my logs go back as far as March 10th and can see the issue there too - becomes unavailable for exactly 30s each time but it occurs randomly and seems to be more frequent when printing rather than just on standby

**Comment by fredrikbaberg** [comment]
I can confirm that my system also goes to "unavailable". I checked this overnight with "binary_sensor.octoprint_printing", each timeout lasts for 30 seconds.

Not sure of the reason for this, it could be the addon itself, OctoPrint integration, or something else.

I'll run some more tests overnight. I won't be able to fix it right now, but hopefully I'll be able to get some more information.

**Comment by fredrikbaberg** [comment]
I can't see any sign that the addon itself stops working. I've let the host ping the addon every 15 seconds for 8 hours without any downtime. The "curl" command I had for verifying that the web interface was responding correctly didn't execute, so I'll have to re-run that test.

However, I suspect the issue is not with the addon but with the integration. The following issue seems similar: https://github.com/home-assistant/core/issues/62319

I'll let the test with "curl" run for a few hours, plus I'm setting up MQTT to see if that behaves the same way.

**Comment by fredrikbaberg** [comment]
I haven't seen any signs of disconnect/unavailable through ping, curl or MQTT, leading me to believe the issue is with the integration as in the topic linked above.

I'll do a final test with a separate install of OctoPrint (separate machine, not addon) and see if I encounter the same.

**Comment by fredrikbaberg** [comment]
Hi,
Good that you've managed to work around the issue with MQTT.

I've been blocked by #290 so I haven't been able to look at a solution to this, but it seems related to the reverse proxy. I have a test version that has been running for 20 hours without any notice in the integration logbook (but there are instead other issues with that one).

**Comment by mwhdc** [comment]
I ran into this issue and found that it was resolved by using a hostname for the OctoPrint host instead of an IP address. This held true for both an OctoPi image and a scripted install on Raspbian Buster.

**Comment by fredrikbaberg** [comment]
I've noticed this happening even using hostname, but only for this add-on, which is the reason I suspect an issue with the reverse proxy.

I'm closing this as I'm working on a new version, available at https://github.com/fredrikbaberg/ha-3dprinter-addons, that should not have the issue.

**Comment by markusressel** [comment]
@fredrikbaberg Hey there, sorry for answering on this closed issue, but I am still facing these periodic dropout issues and I am wondering whether there is a way to include the improvements you made in the new repository within the home assistant integration for octoprint? The reason I am asking this is that I do not use hassos but run a docker instance on my own cluster at home, while octoprint runs on a separate Raspberry Pi 4 within the local network. The current Home Assistant integration works alright, but it would be nice to have these issues fixed for other installation types as well.

Thx for the work!

**Comment by fredrikbaberg** [comment]
Sorry, the changes I did are not applicable to the Home-Assistant integration itself.
I did a lot of changes, e.g. base image and reverse proxy server, but did not touch the code itself. So, while it should resolve the issue I had with the previous add-on, it probably won't help with other installs.
